### ISSUE TITLE
Android debug build takes a very long time for unknown reason

### ISSUE THREAD

**Issue by Trilion99** [korejan/ALVR#46]
Hi, I'm building the ALXR client with `cargo apk build -p alxr-client-quest` on Windows with latest the Android SDK and NDK.
My build "hangs" for more than two hours at 270/274.

`-vv` output (last 5 lines):
```
[alxr-engine-sys 0.10.0] cargo:rerun-if-changed=C:\Users\LP-Research\source\repos\alxr\alvr\openxr-client\alxr-engine-sys\cpp/ALVR-OpenXR-Engine\src\version.cmake
[alxr-engine-sys 0.10.0] cargo:rerun-if-changed=C:\Users\LP-Research\source\repos\alxr\alvr\openxr-client\alxr-engine-sys\cpp/ALVR-OpenXR-Engine\src\version.gradle
[alxr-engine-sys 0.10.0] cargo:rerun-if-env-changed=ALXR_CMAKE_GEN
[alxr-engine-sys 0.10.0] cargo:rerun-if-env-changed=ALXR_BUNDLE_FFMPEG_INSTALL_PATH
    Building [=======================> ] 270/274: alxr-engine-sys(build)
```

No errors are reported. Does anyone have an idea what is taking so long at this point? Any way to make this more verbose than `-vv`?
The build does successfully finish after 2-3 hours.

Thanks,
Klaus

**Comment by korejan** [comment]
@Trilion99 Hey, yeah this issue only happens when building on windows, I'm not 100% sure but it is either an issue with cargo-apk or the way I'm invoking gradle scripts from rust build script causes cargo-apk/build to hang waiting on jdk compiler process to exit and never does.

If you would like to still build on windows, there are 2 workarounds you can do, either:

Open `openxr-client\alxr-engine-sys\android` in android studio, pick the appriopriate "active build variant", build once and leave android-studio open and invoke `cargo apk/xtask`

Or

Once the build gets to the point of hanging, open task manager and search for `OpenJDK Platform binary` process (or which jdk compiler you maybe using) and when you see 0% CPU activity kill the process.

**Comment by korejan** [comment]
I should probably add if you didn't know about it already, there are nightly builds [here](https://github.com/korejan/ALXR-nightly/releases)

**Comment by Trilion99** [comment]
@korejan Thank you for the swift reply! That is great to know, I will do the separate build in Android Studio. I usually do that for the main ALVR branch client, but wasn't sure if it's possible for your variant as well as the structure has changed a bit.

I'm trying to connect Quest, Focus 3 to external tracking systems like Optitrack, ART etc. Therefore I'm trying different modifications of the code and need to rebuild. I could try to build on wsl just for the client but it's more convenient to have everything in one os for now.

**Comment by Trilion99** [comment]
Actually I'll leave this open, if it's ok for you. I saw that someone else had the same issue, but the solution is in the solved issues and I didn't see it.

**Comment by Trilion99** [comment]
@korejan Hmm killing the JDK task works, building in Android Studio didn't help so far. Maybe I'm using the wrong build variant, but tried all Quest related. Could you point me to the part in the cargo apk build script where you invole the gradle scripts? i'll try to dig into the issue a bit.

**Comment by korejan** [comment]
@Trilion99 For android-studio maybe you need to run "Refresh Linked C++ Projects" in the build menu. The cargo/rust build script is `alvr\openxr-client\alxr-engine-sys\build.rs`

For the Focus 3, ignore the wiki about installing the runtime-broker it is not needed for devices that have system installed OpenXR runtimes (that are compatible with android OXR loader). You only need to build/install the generic android flavour of the apks `alxr-client(-debug).apk` and if it wasn't clear in the wiki you don't need Monado, focus 3 has an OpenXR runtime.

**Comment by Trilion99** [comment]
@korejan Thanks for the heads-up. I will test on Focus 3 today and then dig into the Android build on Windows a bit more. In case I manage to find an improvement I'll create a pull-request.
